DNF Handle Empty AppStream stream definition

maxamillion opened the pull request:
Signed-off-by: Adam Miller <admiller@redhat.com>

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Previously, the logic checking for a `stream` definition from a module/AppStream provided wasn't properly handling the scenario in which there was a missing/empty `stream`.

Fixes #63683

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
dnf

maxamillion committed:
DNF Handle Empty AppStream stream definition

Fixes #63683

Signed-off-by: Adam Miller <admiller@redhat.com>

mkrizek reviewed (commented):
LGTM. There seems to be an issue with the testing AppStream `'@ripgrep/default'` in shippable though.

maxamillion commented:
@mkrizek this is interesting, that works on Fedora 30 and 31 but not on Fedora 29. Investigating.

maxamillion committed:
Switch Fedora dnf test target modularity to stratis

In Fedora 29, the metadata was not properly set for a default stream
for ripgrep even though there is a profile called "default", however
that's an arbitrary string and the module maintainer must set the
default stream (which it never was for the ripgrep module, thus
failing the "empty stream" install test)

Signed-off-by: Adam Miller <admiller@redhat.com>

maxamillion commented:
rebuild_merge
